src/main/java/de/philippkatz/maven/plugins/TestParserMojo.java

```diff
@@ -86,6 +86,41 @@ public void execute() throws MojoExecutionException {
 			throw new MojoExecutionException("There were test failures.");
 		}
 	}
+	
+	/**
+	 * Walks over a tree of directories and recursively adds all these directories to a list of <code>File</code> objects.
+	 * 
+	 * @param rootDirectory The root directory from where to start the search for sub directories.
+	 * @return A <code>List</code> of the directory provided as root and all its sub directories.
+	 * @throws MojoExecutionException 
+	 */
+	List<File> collectReportDirectoriesRecursively(final File rootDirectory) throws MojoExecutionException {
+		if(rootDirectory == null) {
+			throw new MojoExecutionException("No valid directory provided");
+		}
+		if(!rootDirectory.exists()) {
+			throw new MojoExecutionException("Directory "+rootDirectory+" does not exist!");
```

Comment:
(1) Coding conventions: spaces around `+` (same applies below)
(2) Consistency: No trailing exclamation mark (same applies below)

Reply:
I did read much about point 2 and usually I try to comply with that. But in this case it would increase nesting and I believe reducing nesting is more important then removing exclamation marks. So our usual convention is, that trailing exclamation marks are ok in the case of guard conditions. According to your own code in the method above, you believe the same.

Reply:
I was talking about the string :-) I didn't use exclamation marks there in the existing code.